lua/lspconfig/server_configurations/swift_mesonls.lua

```diff
@@ -4,7 +4,7 @@ return {
   default_config = {
     cmd = { 'Swift-MesonLSP', '--lsp' },
     filetypes = { 'meson' },
-    root_dir = util.root_pattern('meson_options.txt', '.git'),
+    root_dir = util.root_pattern('meson_options.txt', 'meson.options', '.git'),
```

Comment:
order matters. if the new file has higher precedence, it should be first in the list.

Reply:
You're not allowed to have both files, so the point seems moot. Happy to submit a follow-up if that's what you want.